File: editor/src/de/fatox/meta/ui/windows/CameraWindow.java
package de.fatox.meta.ui.windows;

import com.badlogic.gdx.graphics.PerspectiveCamera;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.kotcrab.vis.ui.widget.VisTextField;
import de.fatox.meta.injection.Inject;
import de.fatox.meta.injection.Singleton;
import de.fatox.meta.ui.components.MetaLabel;

/**
 * Created by Frotty on 20.05.2016.
 */
@Singleton
public class CameraWindow extends MetaWindow {
    private final VisTextField xPosField;
    private final VisTextField yPosField;
    private final VisTextField zPosField;

    private final VisTextField xUpField;
    private final VisTextField yUpField;
    private final VisTextField zUpField;

    @Inject
    public PerspectiveCamera camera;

    public CameraWindow() {
        super("Camera", true, true);
        xPosField = new VisTextField("0.0");
        yPosField = new VisTextField("0.0");
        zPosField = new VisTextField("0.0");

        xUpField = new VisTextField("0.0");
        yUpField = new VisTextField("0.0");
        zUpField = new VisTextField("0.0");

        contentTable.add(new MetaLabel("Position:", 14)).colspan(6).center().row();
        contentTable.add(new MetaLabel("x:", 12));
        contentTable.add(xPosField).width(64).pad(2);
        contentTable.add(new MetaLabel("y:", 12));
        contentTable.add(yPosField).width(64).pad(2);
        contentTable.add(new MetaLabel("z:", 12));
        contentTable.add(zPosField).width(64).pad(2);
        contentTable.row();

        contentTable.add(new MetaLabel("Up:", 14)).colspan(6).center().row();
        contentTable.add(new MetaLabel("x:", 12));
        contentTable.add(xUpField).width(64).pad(2);
        contentTable.add(new MetaLabel("y:", 12));
        contentTable.add(yUpField).width(64).pad(2);
        contentTable.add(new MetaLabel("z:", 12));
        contentTable.add(zUpField).width(64).pad(2);
    }


    @Override
    public void draw(Batch batch, float parentAlpha) {
        super.draw(batch, parentAlpha);
        xPosField.setText(camera.position.x + "");
        yPosField.setText(camera.position.y + "");
        zPosField.setText(camera.position.z + "");

        xUpField.setText(camera.up.x + "");
        yUpField.setText(camera.up.y + "");
        zUpField.setText(camera.up.z + "");
    }
}

File: editor/src/main/kotlin/de/fatox/meta/ui/windows/CameraWindow.kt
package de.fatox.meta.ui.windows

import com.badlogic.gdx.graphics.PerspectiveCamera
import com.badlogic.gdx.graphics.g2d.Batch
import de.fatox.meta.injection.MetaInject.Companion.lazyInject
import de.fatox.meta.ui.components.MetaLabel
import de.fatox.meta.ui.components.MetaTextField

/**
 * Created by Frotty on 20.05.2016.
 */
class CameraWindow : MetaWindow("Camera", true, true) {
	private val xPosField: MetaTextField
	private val yPosField: MetaTextField
	private val zPosField: MetaTextField
	private val xUpField: MetaTextField
	private val yUpField: MetaTextField
	private val zUpField: MetaTextField

	val camera: PerspectiveCamera by lazyInject()

	override fun draw(batch: Batch, parentAlpha: Float) {
		super.draw(batch, parentAlpha)
		xPosField.text = camera.position.x.toString() + ""
		yPosField.text = camera.position.y.toString() + ""
		zPosField.text = camera.position.z.toString() + ""
		xUpField.text = camera.up.x.toString() + ""
		yUpField.text = camera.up.y.toString() + ""
		zUpField.text = camera.up.z.toString() + ""
	}

	init {
		xPosField = MetaTextField("0.0")
		yPosField = MetaTextField("0.0")
		zPosField = MetaTextField("0.0")
		xUpField = MetaTextField("0.0")
		yUpField = MetaTextField("0.0")
		zUpField = MetaTextField("0.0")
		contentTable.add(MetaLabel("Position:", 14)).colspan(6).center().row()
		contentTable.add(MetaLabel("x:", 12))
		contentTable.add(xPosField).width(64f).pad(2f)
		contentTable.add(MetaLabel("y:", 12))
		contentTable.add(yPosField).width(64f).pad(2f)
		contentTable.add(MetaLabel("z:", 12))
		contentTable.add(zPosField).width(64f).pad(2f)
		contentTable.row()
		contentTable.add(MetaLabel("Up:", 14)).colspan(6).center().row()
		contentTable.add(MetaLabel("x:", 12))
		contentTable.add(xUpField).width(64f).pad(2f)
		contentTable.add(MetaLabel("y:", 12))
		contentTable.add(yUpField).width(64f).pad(2f)
		contentTable.add(MetaLabel("z:", 12))
		contentTable.add(zUpField).width(64f).pad(2f)
	}
}
